### ISSUE TITLE
dsymutil generates DWARF units with version 0

### ISSUE THREAD

**Issue by glandium** [llvm/llvm-project#62874]
This is a regression from bd0dd27bb5be0fbf60c1b2a4ce15188812388574.

STR:
- Download and unpack https://drive.google.com/file/d/1ZLiYpu3mFl4W6Qi_a5xfeu2E5LI41s3Y/view?usp=sharing
- run `dsymutil --use-reproducer dsymutil-b354fe /tmp/gecko/obj-x86_64-apple-darwin/toolkit/library/build/XUL -o XUL.dSYM`
- run `llvm-dwarfdump XUL.dSYM > /dev/null`

You will see the following warning:
```
warning: DWARF unit at offset 0x313d661e has unsupported version 0, supported are 2-5
```

This causes problems with some other tools.

It's worth noting that none of the inputs are DWARF v5. They're all DWARF v4.

Cc: @avl-llvm

**Comment by avl-llvm** [comment]
Thank you for reporting the issue. I have reproduced it and is working on the fix.

**Comment by avl-llvm** [comment]
https://reviews.llvm.org/D151348

**Comment by avl-llvm** [comment]
@glandium Would you mind checking whether the problem is fixed by [D151348](https://reviews.llvm.org/D151348), please?

**Comment by glandium** [comment]
It seems to be fixed.
